### ISSUE TITLE
Adding `NOT NULL GENERATED` column triggers warning `adding-required-field`

### ISSUE THREAD

**Issue by peroxy** [sbdchd/squawk#396]
I have this SQL migration:
```
migrations/sql/migration.sql:25:2: warning: adding-required-field

  39 | ALTER TABLE foo
  40 |     ADD COLUMN bar numeric GENERATED ALWAYS AS (bar + baz) STORED NOT NULL;

  note: Adding a NOT NULL field without a DEFAULT will fail for a populated table.
  help: Make the field nullable or add a non-VOLATILE DEFAULT (Postgres 11+).
  ```
  
  This column does not require a `DEFAULT` because it will never be NULL (bar+baz columns are also NOT NULL).

I do not think this warning should be triggered if you have a GENERATED STORED column.

**Comment by chdsbd** [comment]
This is definitely a bug in the `adding-required-field` rule, but I think adding a generated column to a large table would be problematic.

If you add a generated column to a table with like 1 million rows, Postgres will rewrite that whole table, taking a lock and blocking queries on that table.

That issue is kind of outlined on this page: https://squawkhq.com/docs/adding-field-with-default/
